### DISCUSSION TITLE
Is there any reason why compiler does not generate == and != operators for records?

### DISCUSSION THREAD

**Discussion by voroninp** [dotnet/csharplang#3997] [Q&A] [answered]
I was surprised to discover that while compiler provides implementation of `Equals` there are no `==` and `!=` operators. Why is it so?

**Comment by alrz** [comment, marked as answer]
It does (maybe not in your version of the compiler). https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEBMBGAsAKBQZgAIE4BjAewQBNCBhQgb10JdwF8g==

**Comment by MgSam** [comment]
To elaborate, in the earlier preview versions of C# 9 shipped in Visual Studio previews, the `==` operators weren't implemented for records yet (because the LDC hadn't yet decided how they would be implemented). I haven't tested but I assume they were in the most recent VS preview.
